packages/espressocash_app/lib/features/balances/data/token_balance_repository.dart

```diff
@@ -0,0 +1,108 @@
+import 'dart:convert';
+
+import 'package:collection/collection.dart';
+import 'package:fast_immutable_collections/fast_immutable_collections.dart';
+import 'package:injectable/injectable.dart';
+import 'package:rxdart/rxdart.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+import '../../accounts/auth_scope.dart';
+import '../../currency/models/amount.dart';
+import '../../currency/models/currency.dart';
+import '../../tokens/token.dart';
+import '../../tokens/token_list.dart';
+import '../models/token_balance.dart';
+
+@Singleton(scope: authScope)
+class TokenBalancesRepository {
+  TokenBalancesRepository(this._storage, this._tokens);
+
+  final SharedPreferences _storage;
```

Comment:
Let's store it in DB instead, SharedPreferences should be used for something small.

Reply:
refactored to local db